ui/lib/pki/addon/routes/configuration.js

```diff
@@ -4,5 +4,48 @@
  */
 
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { hash } from 'rsvp';
+import { withConfig } from 'pki/decorators/check-config';
 
-export default class PkiConfigurationRoute extends Route {}
+@withConfig()
```

Comment:
nice job moving this out from `configuration/index.js` to the parent level.

Reply:
It made sense where you had it before! But now that there's a child route sharing some resources, I moved it up. I wondered about splitting things up a bit here so only the shared data is in this route (crls and url fetches) which would mean moving `hasConfig` back down to the index. 

The pro would be we'd only be passing necessary data, but required model hooks in both the `pki/configuration/index.js` and `pki/configuration/edit.js` to feel a bit piecemeal. But this seemed like it might make it harder to track data down later, because we'd have to reference both the parent's model and child's - what are your thoughts?

Reply:
I just realized this - by adding this data (`crls and urls`) in the configuration parent then the tidy and create pages will also try to fetch that data? If that's the case, I'm wondering if it should be shared with all the configuration pages since create + tidy doesn't need all of this data?

Reply:
They won't re-fetch the data, but yes the data is passed down. Which is why I was considering intercepting in the model hooks and only passing the necessary data. However, then it got a little hard to follow which data was coming from where. For example, you'd have to look at the configuration parent route beside the child route to look at the combined model hook data. So I opted to keep it all in the parent, since it's not a ton of extra data. But open to considering alternatives!

Reply:
@kiannaquach ...after all that back and fourth I ended up adding the model hooks back to the child routes 😅. A test was failing because of the issuerModel created just for capabilities and when I revisited this, I realized it actually was easier to track down the data when each route had it's model hook clearly defined. Since I changed my mind, I wanted to follow up and let you know!